pkg/operator/ceph/object/user/controller.go

```diff
@@ -571,10 +571,8 @@ func (r *ReconcileObjectStoreUser) validateUser(u *cephv1.CephObjectStoreUser) e
 	if u.Namespace == "" {
 		return errors.New("missing namespace")
 	}
-	if !r.cephClusterSpec.External.Enable {
```

Comment:
This should really be checked for both external and local clusters? Seems like external cluster wouldn't expect u.Spec.Store to be set.

Reply:
Sorry I don't understand that, is there any specific reason for that? For the external rgw server, we do have CephObjectStore() CR right?

Reply:
I guess the question in my mind is whether we should be ignoring the CephCluster's external configuration from the reconcile. From what I gather, this makes it so we only care about the external status of the CephObjectStore. I'm still considering the overall ramifications this might have, positively or negativeluy.

Reply:
I am having concerns about the number of external permutations that exist for CephCluster and CephObjectStore.

With code released to date, I believe we have 3 operating permutations:
1. Local CephCluster with Local CephObjectStore
2. External-mode CephCluster with Local CephObjectStore (mons and osds are remote, but we run the RGW inside k8s) (https://rook.github.io/docs/rook/v1.9/CRDs/Cluster/external-cluster/#cephcluster-example-management)
3. External-mode CephCluster with external-mode CephObjectStore

I think all 3 of these operating permutations make sense. But I think that this PR may be opening up a 4th operating permutation, and I'm not sure this 4th operating mode makes sense as a supportable configuration:
4. Local CephCluster with external-mode CephObjectStore

Reply:
> I am having concerns about the number of external permutations that exist for CephCluster and CephObjectStore.
> 
> With code released to date, I believe we have 3 operating permutations:
> 
> 1. Local CephCluster with Local CephObjectStore
> 2. External-mode CephCluster with Local CephObjectStore (mons and osds are remote, but we run the RGW inside k8s) ([doc](https://rook.github.io/docs/rook/v1.9/CRDs/Cluster/external-cluster/#cephcluster-example-management))
> 3. External-mode CephCluster with external-mode CephObjectStore
> 
> I think all 3 of these operating permutations make sense. But I think that this PR may be opening up a 4th operating permutation, and I'm not sure this 4th operating mode makes sense as a supportable configuration: 4. Local CephCluster with external-mode CephObjectStore

This comment really make sense, what I was talking about

Reply:
I believe external mode was originally created with case 2 working, and I believe it will still work today, though we certainly don't have any CI tests that validate this.

Reply:
@travisn @BlaineEXE @parth-gr Actually, this check has nothing to do with the different modes mentioned, here it validates `cephobjectstoreuser`  CR. There is nothing special for external or local rgw while creating/provisioning `cephobjectstoreUser` both does it with help of adminops API. So IMO to include a check for both.

@BlaineEXE: I not trying to create a different mode not in this PR for sure. And I guess mode 2 should work today not sure whether it is desired or not. What in my mind was to remove the dependency for `Cephcluster` CR for the external rgw server, again not with this PR. I felt External `CephObjectStore CR` can work without knowing ceph details or running ceph commands because Rook considers rgw as an endpoint for the external use case.

Reply:
@thotz this makes sense. but maybe to avoid "option 4" (at least in this PR) we should also make sure that the cluster is external as well?

Reply:
@yuvalif : Actually, there is a bug in the current implementation 
Even if we check for the cluster is external, we cannot make sure the rgw server belongs to that external ceph cluster.
Say I have created `cephcluster` CR with external ceph cluster A and with current checks I can still create `cephObjectStore` CR for rgw server which belongs to another external ceph cluster B.
The plan was to fix it with help https://github.com/ceph/go-ceph/pull/616